### ISSUE TITLE
Eventcalendar format / listen to `$wgLocaltimezone`

### ISSUE THREAD

**Issue by kghbln** [SemanticMediaWiki/SemanticResultFormats#205]
### Setup

- MediaWiki 	1.27.1 (36b636d) 19:26, 11. Jan. 2017
- PHP 	5.6.30-0+deb8u1 (apache2handler)
- MariaDB 	10.0.29-MariaDB-0+deb8u1
- Semantic MediaWiki	2.5.0 (dbbf3d1) 20:58, 14. Mär. 2017
- Semantic Result Formats	2.4.2 (dc464a1) 11:20, 25. Feb. 2017

### Issue

The eventcalendar format ignores settings for `$wgLocaltimezone` like `$wgLocaltimezone = 'Europe/Berlin';` and refers to UTC when displaying information.

### Example
I have setup some [data](https://sandbox.semantic-mediawiki.org/wiki/Event_calendar_data) which [result in this display](https://sandbox.semantic-mediawiki.org/wiki/Event_calendar). A daily event thus shows up a day earlier due to the time shift.

A solution would be to annotate every event for standard and summer time and query one or the other depending if you are on daylight savings but this does not really sound like an appealing solution to me.

**Comment by mwjames** [comment]
> The eventcalendar format ignores settings for $wgLocaltimezone like $wgLocaltimezone = 

I think the `IntlTimeFormatter` (in core) doesn't know the existence of `wgLocaltimezone` or its meaning towards the display of date related values. Extending the display that recognizes `wgLocaltimezone` would be a something that needs to be addressed in core (unless I'm missing something on this very issue) rather than specific to this result printer.

**Comment by kghbln** [comment]
> I think the IntlTimeFormatter (in core) doesn't know the existence of wgLocaltimezone or its meaning towards the display of date related values.

This may very well explain things. The reason why I came up with this is that an event for the whole of e.g. 02.04.2017 is sometimes displayed at 01.04.2017 or 02.04.2017 depending if you are on daylight savings or not. This is what makes things difficult and why I brought `wgLocaltimezone` into the discussion since also `wgLocalTZoffset` does not help at all either.

**Comment by mwjames** [comment]
> This is what makes things difficult and why I brought wgLocaltimezone into the discussion since also wgLocalTZoffset does not help at all either.

I'd say move the discussion to SMW-core but I cannot promise that I will look into this within the next release or so since the `TimeValue` is just so bloated and un-maintainable that every time we change something to it, it is not unlikely that we introduce some sort of regression (despite having extended coverage for the time output).

Maybe it is only a matter of adjusting the `TimeValueFormatter` [0] but since I can't say that from afar it remains a pure conjecture.

[0] https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/src/DataValues/ValueFormatters/TimeValueFormatter.php

**Comment by kghbln** [comment]
> I'd say move the discussion to SMW-core

I do not think that an extra issue should be opened. If yes, just poke me to do so. We could use this one as a reference.

> Maybe it is only a matter of adjusting the TimeValueFormatter [0] but since I can't say that from afar it remains a pure conjecture.

This would be cool however I have the feeling that there is probably more to the issue.
